### ISSUE TITLE
some common way to embed values across Defined / Intersect would be nice

### ISSUE THREAD

**Issue by robx** [i-am-tom/holmes#6]
TLDR: I can use `pure` to get a value to test against into `Defined`, and `singleton` for `Intersect`, but there's nothing polymorphic. Would be nice to have something.

The story behind it:

I wrote a solver for a puzzle type using `Defined`, which worked out fine but turned out pretty slow. So I thought I'd try out `Intersect` to see if that improved things. In what may have been a mistaked I tried to generalize the constraints to be able to use both `Defined` and `Intersect`.

I ran into a couple issues there:

- I had used `over (fmap f)` (because `(.$)` was broken when I started with this), which worked fine due to the `Functor` instance on `Defined`, but `Intersect` doesn't have that. Switching to `(.$)` turned out to work at the expense of having to specify terrifying type signatures. I suppose `Intersect` doesn't naturally admit a `Functor` instance? Otherwise that would be nice to have.
- I used some equality checks against given values with `x .== lift (pure v)`, which caused an `Applicative` constraint which again works for `Defined` but not `Intersect`. That's this ticket.

I'll have to disclaim this report by being very much unsure what I'm doing here -- I'm plugging things together getting them to work rather than really understanding how this stuff should function, so there's a good chance I'm asking for things that don't make sense.

**Comment by i-am-tom** [comment]
Sorry for the super slow response on this and #7 - the coronavirus has been wreaking havoc, and I've been too busy to do much beyond Twitter this week 😥 I'll add this tomorrow and do a new release!

**Comment by i-am-tom** [comment]
I have a fix :) Please feel free to take a look at 1d63c7e and tell me what you think - if it's cool with you, I'll do a release in a couple hours!

**Comment by robx** [comment]
It reads fine. Thanks for looking into this!
